### PR TITLE
Add SideNavComponent

### PR DESCRIPTION
**Pull request by Simonpedro**
- Add SideNavComponent
- Update ActionMenuComponent to allow passing classes to the popover container

https://app.asana.com/0/1203925124186299/1205174172572900/f

image

### CODE REVIEW

**Comment by Simonpedro** [comment]
@tysongach, how do I update the version? You run `npm version` and then amend the commit changing the gem version too?

**Comment by tysongach** [comment]
@Simonpedro I just manually update version number in those two places (I don’t use `npm version`).

File: app/components/anchor/action_menu_component.rb [hunk 1]
```diff
@@ -1,22 +1,24 @@
...
+        **kwargs
       )
-    end
+    }
     renders_many :items
 
-    def initialize
+    def initialize(popover_classes: '', **kwargs)
       @menu_id = self.class.generate_id
+      @popover_classes = popover_classes
```
Comment (tysongach) [inline app/components/anchor/action_menu_component.rb:14]: Thoughts on naming this `menu_classes` to match `menu_id`?

```suggestion
      @menu_classes = menu_classes
```

File: app/components/anchor/side_nav_component.html.erb [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+          "flex-none flex items-center justify-between p-6",
+          "[.side-nav-open_&]:px-4 transition-[padding] duration-500",
+        ),
+      ) do %>
+        <%= inline_svg_tag(
+          "logo.svg",
+          aria: true,
+          data: { toggle_target: "inertify" },
+          size: "48",
+          title: "Buoy", # TODO: Improve
```
Comment (tysongach) [inline app/components/anchor/side_nav_component.html.erb:33]: Buoy is probably fine alt text here…

```suggestion
          title: "Buoy",
```

File: app/components/anchor/side_nav_component.html.erb [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+        ) %>
+
+        <%= tag.button(
+          data: { action: "click->toggle#toggle" },
+          type: "button",
+        ) do %>
+          <%= tag.span(
+            "Hide nav",
```
Comment (tysongach) [inline app/components/anchor/side_nav_component.html.erb:41]:
```suggestion
            t(".hide_nav"),
```

File: app/components/anchor/side_nav_component.html.erb [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+
+          <%= render Anchor::IconComponent.new(
+            classes: "opacity-60 [.side-nav-open_&]:rotate-180 transition-transform duration-500",
+            icon: "fast-left-circle",
+          ) %>
+        <% end %>
+      <% end %>
+
+      <%= tag.nav(
+        aria: { label: "" }, # TODO
```
Comment (tysongach) [inline app/components/anchor/side_nav_component.html.erb:54]: What would be a good label for this nav? ‘Apps’?

File: app/components/anchor/side_nav_component.html.erb [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+
+          <% apps.each do |app| %>
+            <%= c.with_item { link_to(app[:label], app[:path], class: "block w-full") } %>
+          <% end %>
+        <% end %>
+      <% end %>
+
+      <% if nav_items.present? %>
+        <%= tag.nav(
+          aria: { label: "" }, # TODO
```
Comment (tysongach) [inline app/components/anchor/side_nav_component.html.erb:79]: What would be a good label for this nav? I think it’s contextual to the selected app, right? So maybe pull in the app name here (e.g. ‘Medical Device’, etc.)?

File: app/components/anchor/side_nav_component.html.erb [hunk 6]
```diff
@@ -0,0 +1,126 @@
...
+                  <%= item[:label] %>
+                <% end %>
+              </li>
+            <% end %>
+          </ul>
+        <% end %>
+      <% end %>
+
+      <%= tag.nav(
+        aria: { label: "" }, # TODO
```
Comment (tysongach) [inline app/components/anchor/side_nav_component.html.erb:102]: What would be a good label for this nav? Maybe ‘User’ if it’s oriented towards user/account navigation?

Comment (tysongach) [reply app/components/anchor/side_nav_component.html.erb:102]: Actually, maybe this should be a `<footer>` instead of `<nav>`?

File: demo/app/views/layouts/component_preview_plain.html.erb
Comment (tysongach) [inline demo/app/views/layouts/component_preview_plain.html.erb:1]: I think we can rename this file to simply `plain.html.erb`.

File: previews/anchor/dialog_component_preview.rb [hunk 7]
```diff
@@ -3,17 +3,24 @@ class DialogComponentPreview < ViewComponent::Preview
     # @param title text
     # @param body_text textarea
     def playground(
-      body_text: "Content",
```
Comment (tysongach) [inline previews/anchor/dialog_component_preview.rb:6]: Why the changes here? Probably best to stay consistent with double instead of partially switching to single. We should get Standard or Rubocop setup in this repo soon…

**Review by tysongach — COMMENTED** [review]
Thanks for pushing this along!

File: app/components/anchor/side_nav_component.html.erb (hunk 4 above)
Comment (Simonpedro) [reply app/components/anchor/side_nav_component.html.erb:54]: 'Apps' seems a good one to me, yes.

File: app/components/anchor/side_nav_component.html.erb (hunk 5 above)
Comment (Simonpedro) [reply app/components/anchor/side_nav_component.html.erb:79]: Yep, that'd work, I could also use `aria-current="page"` for the selected nav item

**Review by tysongach — APPROVED** [review]
🚀

**Comment by Simonpedro** [comment]
@tysongach Just pubished the npm package, could you publish the gem? I don't think I'm registered in https://rubygems.org
